src/GdbConnection.cc

```diff
@@ -1483,6 +1504,14 @@ void GdbConnection::notify_stop(GdbThreadId thread, int sig,
                                 const char *reason) {
   DEBUG_ASSERT(req.is_resume_request() || req.type == DREQ_INTERRUPT);
 
+  // don't pass this signal to gdb if it is specified not to
+  if (pass_signals.find(to_gdb_signum(sig)) != pass_signals.end()) {
+    LOG(debug) << "discarding stop notification for signal " << sig 
+                << " on thread " << thread << " as specified by QPassSignal";
+
+    return;
```

Comment:
In addition to this, you want to modify `GdbServer::debug_one_step`'s `stop_filter` to ignore signal stops that won't be passed to the debugger. This will make reverse execution through ignored signal-stops *much* faster.

Reply:
I added the check to the stop_filter however it didn't yield significant performance improvements on a simple benchmark. Is there a better way to test the performance from this addition?